src/java.base/share/classes/sun/security/util/AbstractAlgorithmConstraints.java

```diff
@@ -41,6 +44,8 @@ public abstract class AbstractAlgorithmConstraints
         implements AlgorithmConstraints {
 
     protected final AlgorithmDecomposer decomposer;
+    private static final Map<String, Pattern> patternCache =
+            new ConcurrentHashMap<>();
```

Comment:
I don't know if this would matter much, but you could probably specify a smaller initial size here, like 4 or 8.

Reply:
Good point, it's very unlikely we'll ever use more than 4 patterns. The default size is 16.